src/data/scripted_effects.rs

```diff
@@ -37,6 +38,9 @@ impl Effects {
             err(ErrorKey::NameConflict).strong().msg(msg).loc(key).push();
         } else {
             let scope_override = self.scope_overrides.get(key.as_str()).copied();
+            if block.source.is_some() {
```

Comment:
This pre-seeding might not be useful and it makes the API of expand_macro more complicated. It made sense when you could say the macro-map was read only during validation, but that's not the case now.

Reply:
Although this is not strictly necessary, it is good to load all the scripted things here rather than later during the validation stage, in order to lower lock contention, especially since it requires a write lock.